File: fund-tracker/backend/app/schemas/returns.py
from pydantic import BaseModel
from datetime import date
from typing import Optional


class ReturnWindowSchema(BaseModel):
    label: str              # "2Y", "3Y", "5Y"
    years: int
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    start_nav: Optional[float] = None
    end_nav: Optional[float] = None
    cumulative_return: Optional[float] = None
    cagr: Optional[float] = None
    data_complete: bool = True
    missing_days: int = 0
    warning: Optional[str] = None


class ReturnsResponse(BaseModel):
    fund_id: int
    latest_nav: Optional[float] = None
    latest_date: Optional[date] = None
    windows: list[ReturnWindowSchema]


class ChartPoint(BaseModel):
    date: date
    nav: float


class ChartResponse(BaseModel):
    fund_id: int
    period: str
    series: list[ChartPoint]

File: fund-tracker/frontend/src/types/returns.ts
export interface ReturnWindow {
  label: string;
  years: number;
  start_date: string | null;
  end_date: string | null;
  start_nav: number | null;
  end_nav: number | null;
  cumulative_return: number | null;
  cagr: number | null;
  data_complete: boolean;
  missing_days: number;
  warning: string | null;
}

export interface ReturnsData {
  fund_id: number;
  latest_nav: number | null;
  latest_date: string | null;
  windows: ReturnWindow[];
}

export interface ChartPoint {
  date: string;
  nav: number;
}

export interface ChartData {
  fund_id: number;
  period: string;
  series: ChartPoint[];
}
